### PR TITLE
Better glue

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

**Media**

Peek 2023-06-16 19-09

**Changelog**

:cl:
- tweak: you can't eat crazy glue anymore but you can drink it
- tweak: space glue tube can now also be used to glue items
- tweak: items are glued for longer duration (10 seconds per unit)
- tweak: glue tubes can be used multiple times and can be refilled with space glue
- remove: removed empty crazy glue

### CODE REVIEW

**Comment by Slava0135** [comment]
I don't know why duration is twice as long, it should be 10 seconds per unit, not 20

**Comment by brainfood1183** [comment]
> I don't know why duration is twice as long, it should be 10 seconds per unit, not 20

this is amazing although i suspect its going to be grief city if the duration can get to long.

**Comment by brainfood1183** [comment]
would be cool if you oculd glue airlocks shut for a duration of time.

**Comment by deltanedas** [comment]
what if instead of glue you used welding fuel and you ignited it to produce an extremely hot flame then used it to seal the noor

**Comment by brainfood1183** [comment]
> what if instead of glue you used welding fuel and you ignited it to produce an extremely hot flame then used it to seal the noor

that could work, although wouldnt be associated with glue i guess. Maybe allow glue to stop bleeding kinda like what glue is used for in army. Glue wound together.

File: Content.Server/Glue/GlueSystem.cs [hunk 1]
```diff
@@ -1,59 +1,69 @@
...
-            }
+        if (TryGlue(uid, component, target))
+        {
+            args.Handled = true;
+            _audio.PlayPvs(component.Squeeze, uid);
+            _popup.PopupEntity(Loc.GetString("glue-success", ("target", Identity.Entity(target, EntityManager))), args.User, args.User, PopupType.Medium);
+        }
+        else
+        {
+            _popup.PopupEntity(Loc.GetString("glue-failure", ("target", Identity.Entity(target, EntityManager))), args.User, args.User, PopupType.Medium);
```
Comment (AJCM-git) [inline Content.Server/Glue/GlueSystem.cs:46]: im pretty sure `THE` fluent function already does identity stuff, so that is innecesary here

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Shared/Glue/GluedComponent.cs [hunk 2]
```diff
@@ -10,33 +9,11 @@ public sealed class GluedComponent : Component
...
-    /// </summary>
-    public bool GlueBroken = false;
+    [DataField("until", customTypeSerializer: typeof(TimeOffsetSerializer)), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan Until = TimeSpan.Zero;
 
-    [DataField("glued")]
-    [ViewVariables(VVAccess.ReadWrite)]
-    public bool Glued = false;
+    [DataField("duration", customTypeSerializer: typeof(TimeOffsetSerializer)), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan Duration = TimeSpan.Zero;
```
Comment (AJCM-git) [inline Content.Shared/Glue/GluedComponent.cs:18]: i think these are causing the test error

**Comment by Slava0135** [comment]
> I don't know why duration is twice as long, it should be 10 seconds per unit, not 20

this happened because of time span serialization stuff
rn glue tube applies effect for 30 seconds and crazy glue - for 50 seconds

**Comment by Slava0135** [comment]
AAAA

**Review by AJCM-git — APPROVED** [review]

File: Content.Shared/Glue/GlueComponent.cs [hunk 3]
```diff
@@ -1,15 +1,42 @@
...
+    /// Reagent consumption per use.
+    /// </summary>
+    [DataField("consumption"), ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 Consumption = FixedPoint2.New(3);
+
+    /// <summary>
+    /// Duration per unit
+    /// </summary>
+    [DataField("duration"), ViewVariables(VVAccess.ReadWrite)]
+    public TimeSpan Duration = TimeSpan.FromSeconds(10);
```
Comment (AJCM-git) [inline Content.Shared/Glue/GlueComponent.cs:41]: Rename to DurationPerUnit, add another datafield capping it to like 60 seconds (so you cant put 7 bottles of glue on something for a de-facto admin smite)  (maybe make it not possible to pour more than the cap and do a popup to have some player feedback)

File: Content.Shared/Glue/GlueComponent.cs [hunk 4]
```diff
@@ -1,15 +1,42 @@
...
+    /// Reagent that will be used as glue.
+    /// </summary>
+    [DataField("reagent", customTypeSerializer: typeof(PrototypeIdSerializer<ReagentPrototype>))]
+    public string Reagent = "SpaceGlue";
+
+    /// <summary>
+    /// Reagent consumption per use.
+    /// </summary>
+    [DataField("consumption"), ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 Consumption = FixedPoint2.New(3);
```
Comment (AJCM-git) [inline Content.Shared/Glue/GlueComponent.cs:35]: rename to ConsumptionUnit to make it more obvious, also make this 5

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Content.Shared/Glue/GlueComponent.cs (hunk 3 above)
Comment (Slava0135) [reply Content.Shared/Glue/GlueComponent.cs:41]: right now you can't put glue on item more than once so it doesn't stack

**Comment by Slava0135** [comment]
normal glue tube glues items for 30 seconds, crazy one - for 60 seconds, you can't reapply effect on already glued item as before

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**